jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/PathMappingsHandler.java

```diff
@@ -79,6 +80,15 @@ public void dump(Appendable out, String indent) throws IOException
         Dumpable.dumpObjects(out, indent, this, mappings);
     }
 
+    @Override
+    protected void doStart() throws Exception
+    {
+        Server server = getServer();
+        for (MappedResource<Handler> mapping: mappings)
+            mapping.getResource().setServer(server);
+        super.doStart();
+    }
+
```

Comment:
Uhm, this is not where other containers do it.
They typically override `setServer()` and also do it in their own specific methods to add children such as `Handler.Sequence.setHandlers()`, etc.
I would do the same, not overriding `doStart()` which may happen too late.

I'd check also if `Singleton.updateHandler()` would do the job.
Currently, `PathMappingsHandler.addMapping()` calls only `addBean()`, but if it calls `Singleton.updateHandler()` it will probably be best?

Reply:
`Singleton.updateHandler()` is not appropriate, as that's for a container that has only 1 handler.

The handlers themselves are also not beans on the PathMappingsHandler.
The PathMappings object is the bean.
And that PathMappings bean holds the Handlers.

The doStart() is still sane, and performs the appropriate doStart() on the handlers.
But there's no connection between the PathMappingsHandler and the PathMappings -> Handlers for the setServer() call.

Note that the `Handler.setServer(Seerver)` call occurs during `.setHandler(Handler)` calls, like the `Server.setHandler(handlers)` call.
So that means if you add the `PathMappingHandler` to the `Server.setServer(handlers)` tree, THEN add handlers to the `PathMappingsHandler` you wind up with no further `Handler.setServer()` calls.

Anyway, I'll move this out from doStart() to addMapping() instead, where it is more appropriate.
Plus I will take the other (now standard) improvements made in other Handlers as well.